internal/discovery/foreign-cluster-operator/setup.go

```diff
@@ -42,10 +43,17 @@ func StartOperator(mgr *manager.Manager, namespace string, requeueAfter time.Dur
 		os.Exit(1)
 	}
 
+	advClient, err := protocolv1.CreateAdvertisementClient(filepath.Join(os.Getenv("HOME"), ".kube", "config"), nil)
```

Comment:
maybe it's remained from debug
if you need an in-cluster client you should leave the kubeconfigPath to ""
if you want to use a specified kubeconfigPath I would pass it as parameter rather than hard-code it